core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/io/AwsChunkedEncodingInputStream.java

```diff
@@ -37,37 +35,18 @@
  * the wrapped stream.
  */
 @SdkInternalApi
-public abstract class AwsChunkedEncodingInputStream extends SdkInputStream {
+public abstract class AwsChunkedEncodingInputStream extends AwsChunkedInputStream {
```

Comment:
This chunked encoding code is getting messy and hard for me to follow. I'm cool with it for now to hit a date, but would it be better at some point to isolate out the parts that deal with processing the chunked encoding from the transformation/compression itself?

Example: https://gist.github.com/millems/b877ac08bed6f5e028b45edcfa6f1786

Reply:
Yeah agreed, the classes are definitely confusing as is. The chunked encoding parts are actually not used at all by the `AwsCompressionInputStream`, since we leave it to the underlying HTTP client. They are used by `AwsUnsignedChunkedEncodingInputStream` (checksums) and `AwsSignedChunkedEncodingInputStream` (signers). Might be better to refactor later as a separate PR since its not related to compression.

To clarify, the parent/child classes look like:
`AwsChunkedInputStream` (base)
- `AwsCompressionInputStream`
- `AwsChunkedEncodingInputStream`
    * `AwsUnsignedChunkedEncodingInputStream`
    * `AwsSignedChunkedEncodingInputStream`
    
We could maybe remove `AwsChunkedInputStream` and have some duplicate code in `AwsCompressionInputStream` and `AwsChunkedEncodingInputStream` to reduce the number of classes